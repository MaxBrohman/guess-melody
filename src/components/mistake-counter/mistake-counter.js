import React from 'react';
// eslint-disable-next-line
export const MistakeCounter = ({mistakes}) => (
  <div className="game__mistakes">
    {// eslint-disable-next-line
        mistakes.map((mistake, i) => <div className="wrong" key={i}>{mistake}</div>)
    }
  </div>
);
