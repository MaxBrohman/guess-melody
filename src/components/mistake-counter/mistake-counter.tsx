import React from 'react';

type TMistakesCounterProps = {
  mistakes: Array<boolean>;
}

export const MistakeCounter = ({mistakes}: TMistakesCounterProps): JSX.Element => (
  <div className="game__mistakes">
    {
      mistakes.map((mistake, i) => <div className="wrong" key={i}>{mistake}</div>)
    }
  </div>
);
