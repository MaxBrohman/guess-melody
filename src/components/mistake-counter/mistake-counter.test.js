import React from 'react';
import renderer from 'react-test-renderer';

import {MistakeCounter} from './mistake-counter';

it(`Renders MistakeCounter component correctly`, () => {
  const tree = renderer
                .create(
                    <MistakeCounter
                      mistakes={new Array(2)}
                    />
                )
                .toJSON();
  expect(tree).toMatchSnapshot();
});
