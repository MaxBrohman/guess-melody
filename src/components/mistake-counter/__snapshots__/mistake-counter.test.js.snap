// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders MistakeCounter component correctly 1`] = `
<div
  className="game__mistakes"
/>
`;
